ChessStatePattern/src/states/State.java

```diff
@@ -1,16 +1,36 @@
 package states;
```

Comment:
Left comments right here,

The PR title need to be modified that clear to appear what's the pr about, not only 'state'
For example, it's related to State class and Move class, the title should mention what's change/add/modify on them.